lib/chef/resource/chef_sleep.rb

```diff
@@ -0,0 +1,69 @@
+#
+# Copyright:: 2019, Chef Software Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require_relative "../resource"
+require_relative "../dist"
+
+class Chef
+  class Resource
+    class ChefSleep < Chef::Resource
+      resource_name :chef_sleep
+      provides :chef_sleep
+
+      unified_mode true
+
+      description "Use the chef_sleep resource to sleep for a number of seconds during a #{Chef::Dist::PRODUCT} run. This resource should only be used when a command or service exits successfuly , but is not fully ready for the next step of the recipe."
+      introduced "15.5"
+      examples <<~DOC
+        Sleep for 10 seconds
+        ```ruby
+        chef_sleep '10'
+        ```
+
+        Sleep for 10 seconds with a friendly resource name for logging
```

Comment:
I don't get the "friendly resource name for logging" part.

Reply:
Instead of the resource name being 10 it's an actual descriptive name